src/openforms/emails/tests/test_digest_functions.py

```diff
@@ -0,0 +1,405 @@
+        logevent.email_status_change(
+            submission,
+            event="registration",
+            status=Message.STATUS_FAILED,
+            status_label="Failed",
+            include_in_daily_digest=True,
+        )
+
+        failed_emails = collect_failed_emails(
+            since=datetime(2023, 2, 2, 13, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(failed_emails, [])
+
+
+@freeze_time("2023-01-02T12:30:00+01:00")
+class FailedRegistrationsTests(TestCase):
+
+    def test_failed_registrations_are_collected(self):
+        # 1st form with 2 failures in the past 24 hours
+        form_1 = FormFactory.create()
+        failed_submission_1 = SubmissionFactory.create(
+            form=form_1, registration_status=RegistrationStatuses.failed
+        )
+        failed_submission_2 = SubmissionFactory.create(
+            form=form_1, registration_status=RegistrationStatuses.failed
+        )
+
+        # 1st failure
+        logevent.registration_failure(
+            failed_submission_1,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        # 2nd failure
+        logevent.registration_failure(
+            failed_submission_2,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        # 2nd form with 1 failure in the past 24 hours
+        form_2 = FormFactory.create()
+        failed_submission = SubmissionFactory.create(
+            form=form_2, registration_status=RegistrationStatuses.failed
+        )
+
+        # failure
+        logevent.registration_failure(
+            failed_submission,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        failed_registrations = collect_failed_registrations(
+            since=datetime(2023, 1, 1, 14, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(len(failed_registrations), 2)
+        self.assertEqual(failed_registrations[0].failed_submissions_counter, 2)
+        self.assertEqual(failed_registrations[1].failed_submissions_counter, 1)
+
+    def test_successful_registrations_are_not_collected(self):
+        form = FormFactory.create()
+        SubmissionFactory.create(form=form, registration_success=True)
+
+        failed_registrations = collect_failed_registrations(
+            since=datetime(2023, 1, 2, 14, 30, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(failed_registrations, [])
+
+    def test_failed_registrations_more_than_24h_are_not_collected(self):
+        form = FormFactory.create()
+        submission = SubmissionFactory.create(form=form, registration_failed=True)
+
+        logevent.registration_failure(
+            submission,
+            RegistrationFailed("Registration plugin is not enabled"),
+        )
+
+        failed_registrations = collect_failed_registrations(
+            since=datetime(2023, 1, 4, 1, 0, 0).replace(tzinfo=utc)
+        )
+
+        self.assertEqual(failed_registrations, [])
```

Comment:
I don't see this 24h aspect in the `collect_failed_registrations` function, it's just querying based on `since`. 

I don't think that this aspect is really testable, it all comes down to "are we passing the right `since`" parameter.

You *could* perhaps do:

```py
digest = Digest(since=datetime(2023, 1, 4, 1, 0, 0).replace(tzinfo=utc))
context = digest.get_context_data()

assert context["failed_emails"] == []
```
which checks that the contraint is passed along properly.